src/core/InformationPanel/InformationPanel.tsx

```diff
@@ -0,0 +1,166 @@
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+import cx from 'classnames';
+import React from 'react';
+import { IconButton } from '../Buttons';
+import { CommonProps, useMergedRefs, useTheme } from '../utils';
+import SvgCloseSmall from '@itwin/itwinui-icons-react/cjs/icons/CloseSmall';
+import '@itwin/itwinui-css/css/information-panel.css';
+
+export type InformationPanelProps = {
+  /**
+   * Label shown at the top of the panel.
+   */
+  label?: React.ReactNode;
+  /**
+   * Is the panel open?
+   * @default false
+   */
+  isOpen?: boolean;
+  /**
+   * Orientation of the panel.
+   * @default 'vertical'
+   */
+  orientation?: 'vertical' | 'horizontal';
+  /**
+   * Callback fired when close icon is clicked.
+   */
+  onClose?: () => void;
+  /**
+   * Custom header action buttons shown alongside the close button.
+   * @example
+   * [
+   *   <IconButton styleType='borderless' onClick={() => {}}><SvgEdit /></IconButton>,
+   *   <IconButton styleType='borderless' onClick={() => {}}><SvgDelete /></IconButton>,
+   * ]
+   */
+  headerActions?: React.ReactNode[];
+  /**
+   * Set to false to make the panel non-resizable.
+   * @default true
+   */
+  resizable?: boolean;
+  /**
+   * Content of the panel.
+   */
+  children?: React.ReactNode;
+} & Omit<CommonProps, 'title'>;
+
+/**
+ * InformationPanel provides a way to view additional content or attributes
+ * without leaving the main content area (e.g. Table).
+ *
+ * The panel appears as an overlay on top of the main content.
+ * @example
+ * <InformationPanelWrapper>
+ *   <Table ... /> // component to overlay
+ *   <InformationPanel
+ *     label={<Text variant='subheading'>InfoPanel heading</Text>}
+ *     onClose={() => {}}
+ *     isOpen={true}
+ *   >
+ *     <span>Info panel content</span>
+ *   </InformationPanel>
+ * </InformationPanelWrapper>
+ */
+export const InformationPanel = React.forwardRef<
+  HTMLDivElement,
+  InformationPanelProps
+>((props, ref) => {
+  const {
+    className,
+    label,
+    isOpen = false,
+    onClose,
+    orientation = 'vertical',
+    headerActions,
+    resizable = true,
+    children,
+    ...rest
+  } = props;
+
+  useTheme();
+
+  const infoPanelRef = React.useRef<HTMLDivElement>(null);
+  const refs = useMergedRefs(ref, infoPanelRef);
+
+  const onResize = React.useCallback(
+    (e: PointerEvent) => {
+      e.preventDefault();
+      if (!infoPanelRef.current) {
+        return;
+      }
+      const { right, bottom } = infoPanelRef.current.getBoundingClientRect();
+      if (orientation === 'vertical') {
+        infoPanelRef.current.style.width = `${right - e.clientX}px`;
+      } else {
+        infoPanelRef.current.style.height = `${bottom - e.clientY}px`;
+      }
+    },
+    [orientation],
+  );
+
+  return (
+    <div
```

Comment:
Shouldn't we use `react-transition-group` here to prevent rendering stuff even when it is not visible? Users might have some heavy content here e.g. map.

Reply:
I'm starting to dislike that library after trying to use it for SideNav submenu 😆 Plus it would require CSS changes.

I am not sure of the performance difference between re-rendering hidden content vs unmounting/remounting. And I think browsers do have the ability to lazy load stuff like iframes (maps) with just one property. Not to mention, the user can add their own performance optimizations.